server/models/service-process.js

```diff
@@ -142,6 +149,11 @@ module.exports = function extendServiceProcess(ServiceProcess) {
     });
 
     function updateProcessStatus(proc, asyncCb) {
+      if (!proc)
+        return asyncCb(Error(
+          util.format('Process state update for unknown process: %j', pInfo)
+        ));
+
       proc.isTracing = !!pInfo.isTracing;
```

Comment:
Unrelated to your PR, but maybe you could delete this line that isn't doing anything?